src/rgw/rgw_rados.cc

```diff
@@ -1518,6 +1520,11 @@ int RGWZoneParams::fix_pool_names()
   return 0;
 }
 
+void RGWZoneParams::set_pool_namespaces()
+{
+  roles_pool.name_space = "role";
+}
+
```

Comment:
To avoid parsing the 'poolname:namespace' string, as you have stated in your earlier comment, I added another method to set the namespaces. And anyway we will be hardcoding the names as poolname:namespace, so i hardcoded the namespace here. It is a less intuitive, but simpler. Let me know what you think.